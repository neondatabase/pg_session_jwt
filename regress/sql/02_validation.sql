-- Validation tests for pg_session_jwt

-- Initialize the session
SELECT auth.init();

-- Test JWT token ID (jti) monotonically increasing
DO $$
DECLARE
    -- These are placeholder values that should be replaced with real JWTs at test time
    test_jwt1 TEXT := 'PLACEHOLDER.JWT.VALUE1';
    test_jwt2 TEXT := 'PLACEHOLDER.JWT.VALUE2';
BEGIN
    PERFORM auth.jwt_session_init(test_jwt1);
    PERFORM auth.jwt_session_init(test_jwt2);
END;
$$;

-- Test JWT payload retrieval
SELECT jsonb_typeof(auth.session()) AS session_type;
SELECT auth.user_id() AS user_id;
