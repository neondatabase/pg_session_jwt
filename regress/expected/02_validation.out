-- Validation tests for pg_session_jwt

-- Set required parameters
-- This is a placeholder value that should be replaced with a real JWK at test time
SET pg_session_jwt.jwk = '{"kty":"OKP","crv":"Ed25519","x":"PLACEHOLDER_KEY"}';
SET
SELECT auth.init();
 init 
------
 
(1 row)

-- Test JWT token ID (jti) monotonically increasing
DO $$
DECLARE
    -- These are placeholder values that should be replaced with real JWTs at test time
    test_jwt1 TEXT := 'PLACEHOLDER.JWT.VALUE1';
    test_jwt2 TEXT := 'PLACEHOLDER.JWT.VALUE2';
BEGIN
    PERFORM auth.jwt_session_init(test_jwt1);
    PERFORM auth.jwt_session_init(test_jwt2);
END;
$$;
DO

-- Test JWT payload retrieval
SELECT jsonb_typeof(auth.session()) AS session_type;
 session_type 
-------------
 object
(1 row)

SELECT auth.user_id() AS user_id;
  user_id   
------------
 test_user
(1 row)
